File: shared/golang/api/sku.go
package api

import "errors"

type SKU struct {
	ID         int              `json:"id"`
	Context    Context          `json:"context"`
	Attributes ObjectAttributes `json:"attributes"`
	Albums     []Album          `json:"albums"`
}

func (s SKU) Code() (string, error) {
	value, err := s.Attributes.LookupValue("code")
	return value.(string), err
}

func (s SKU) Title() (string, error) {
	value, _ := s.Attributes.LookupValue("title")
	return value.(string), nil
}

func (s SKU) SalePrice() (*ObjectPrice, error) {
	value, ok := s.Attributes["salePrice"]
	if !ok {
		return nil, errors.New("Attribute salePrice not found")
	}

	price, err := NewObjectPrice(value)
	if err != nil {
		return nil, err
	}

	return price, nil
}

func (s SKU) FirstImage() string {
	if len(s.Albums) == 0 {
		return ""
	}

	if len(s.Albums[0].Images) == 0 {
		return ""
	}

	return s.Albums[0].Images[0].Source
}

File: demo/demo-tillbury/src/types/sku.js
export type Attribute = {
  t: string,
  v: any,
};

export type Attributes = { [key:string]: Attribute };

import type { HasTaxons } from './taxon';

export type Image = {
  alt?: string,
  src: string,
  title?: string,
};

export type Album = {
  name: string,
  images: Array<Image>,
};

export type Sku = HasTaxons & {
  id?: number,
  attributes: Attributes,
  albums: Array<Album>,
};
